src/Components/ProductTable.jsx

```diff
@@ -45,6 +45,26 @@ export default function ProductTable({ cart, updateCart }) {
     setProducts(body);
   }, []);
 
+  useEffect(() => {
+    let sortedProducts = [...products];
+
+    for (let i = 0; i < sortOptions.length; i++) {
+      if (sortOptions[i].current === true) {
+        if (sortOptions[i].name === "Price") {
+          console.log("sorting by price");
+          sortedProducts.sort(
+            (itemA, itemB) => Number(itemB.price) - Number(itemA.price)
+          );
+        } else {
+          sortedProducts.sort(
+            (itemA, itemB) => itemB.releaseDate - itemA.releaseDate
+          );
+        }
+      }
+    }
+    setProducts((prevState) => sortedProducts);
+  }, [sortOptions]);
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
**praise:** ‏Good work tying this logic to changes to the sortOptions dependency.